apps/showcase/src/components/training/code-editor-view/code-editor-view.component.ts

```diff
@@ -167,23 +177,71 @@ export class CodeEditorViewComponent implements OnDestroy, OnChanges {
       file: ''
     });
 
+  /**
+   * Subject used to notify when monaco editor has been initialized
+   */
+  public readonly monacoReady = new Subject<void>();
+
+  /**
+   * Promise resolved with the global monaco instance
+   */
+  private readonly monacoPromise = firstValueFrom(this.monacoReady.pipe(
+    map(() => window.monaco)
+  ));
+
   /**
    * Configuration for the Monaco Editor
    */
   public editorOptions$ = this.form.controls.file.valueChanges.pipe(
     startWith(''),
     filter((filePath): filePath is string => !!filePath),
-    map((filePath) => ({
+    map(() => ({
       theme: 'vs-dark',
-      language: editorOptionsLanguage[filePath.split('.').pop() || 'ts'] || editorOptionsLanguage.ts,
-      readOnly: (this.editorMode === 'readonly'),
+      readOnly: (this.editorMode() === 'readonly'),
       automaticLayout: true,
       scrollBeyondLastLine: false,
-      overflowWidgetsDomNode: this.monacoOverflowWidgets.nativeElement
+      overflowWidgetsDomNode: this.monacoOverflowWidgets.nativeElement,
+      model: this.model()
     }))
   );
 
+  private readonly fileContentLoaded$ = this.form.controls.file.valueChanges.pipe(
+    takeUntilDestroyed(),
+    combineLatestWith(this.cwdTree$),
+    filter(([path, monacoTree]) => !!path && checkIfPathInMonacoTree(monacoTree, path.split('/'))),
+    switchMap(([path]) => from(this.webContainerService.readFile(`${this.project().cwd}/${path}`).catch(() => ''))),
+    share()
+  );
+
+  private readonly fileContent = toSignal(this.fileContentLoaded$);
+
+  /**
+   * Model used for monaco editor for the currently selected file.
+   * We need that to associate the opened file to a URI which is necessary to resolve relative paths on imports.
+   */
+  public model = computed(() => {
+    const value = this.fileContent();
+    const fileName = this.form.controls.file.value!;
+    const fileExtension = fileName.split('.').at(-1);
+    return {
+      value,
+      language: editorOptionsLanguage[fileExtension || ''] || '',
+      uri: `file:///${fileName}`
+    };
+  });
+
   constructor() {
+    effect(async () => {
+      const project = this.project();
+      await untracked(async () => {
```

Comment:
is the `untracked` necessary here?
I suppose it is used for the `this.loadNewProject()` but it also use only `this.project()` an already tracked signal.

Reply:
it's not necessary, but I'm only interested in tracking `this.project()` so I've put it there to make sure we don't track anything else by mistake